### DISCUSSION TITLE
Persisting metadata across plugin runs (utilising the cache? 🤔 )

### DISCUSSION THREAD

**Discussion by meyer** [babel/babel#13590] [Q&A] [answered]
_moved from babel Slack to GitHub Discussions™_

Is there a nice plugin-accessible way to share and persist metadata across runs of babel? I know babel has a cache but I can’t figure out how to read from and write to it (without using `fs.readFile`/`writeFile` of course).

I’ve written a webpack plugin that runs a babel transform and I’m sharing metadata between babel instances with a webpack loader + [loader context](https://github.com/jsxstyle/jsxstyle/blob/dbdfe74/packages/jsxstyle-webpack-plugin/src/loader.ts#L48) + [some fs shenanigans](https://github.com/jsxstyle/jsxstyle/blob/dbdfe74/packages/jsxstyle-webpack-plugin/src/plugin.ts#L54-L57), but I’d like to extract the transform out as a babel plugin so it can be used in less webpack-y environments (for example, plain babel or rollup with rollup-plugin-babel).

In an ideal world the cache would be in memory and only written to disk in some kind of lifecycle event. I really don’t want to read and write one time per babel’d file 😰

**Comment by nicolo-ribaudo** [comment, marked as answer]
If you want the same cache for every transformed file, would a simple global `WeakMap` (defined right outside from the function plugin) work?

**Reply by meyer** [reply]
That would solve part of the problem and would allow builds to be deterministic for as long as the node process sticks around.

The final missing piece would be to persist the data to disk. I could probably do some `process.on` trickery but I’d rather use something official if something official exists.

**Reply by nicolo-ribaudo** [reply]
Unfortunately we don't provide an API like that to plugins :sweat_smile:

**Reply by meyer** [reply]
Shoooot, that’s too bad. Thanks for the help.
